docs/checklist.md

```diff
@@ -27,6 +27,48 @@ Comes in three sections. You need varying degrees of access depending on where y
 - ChecklistProvider: Wraps around the carousel so that the toggle buttons in the secondary menu know when the checklist is open and the other popup menus there (help center) can respect it. 
 - ChecklistCountProvider, ChecklistCategoryProvider: Wrap around various parts of the checklist itself to provide counts of issues.
 
+## Roles
```

Comment:
This section reminds me of #7851 and the discussion around that.

Note that we're planning on making these kind of capability checks much easier, including documentation, see #8275.

Generally speaking, it's important to think about these as individual permissions (capabilities) that need to be accounted for. User roles are just a group of capabilities and roles can vary between sites.

----

For each checklist issue I'd ask "Which permission does the user need to resolve this issue?".

Examples:

* `Story missing title` -> The user needs to be able to edit the current story -> They are already editing the story, so they obviously have permission for that -> no extra check needed
* `Video element missing captions` -> The user needs to upload media (a captions file) to fix this -> check for `hasUploadMediaAction`
* `Video not optimized` -> The user needs to upload the optimized video to the server -> check for `hasUploadMediaAction`
* `Page has too many links` -> The user needs to be able to edit the current story -> They are already editing the story, so they obviously have permission for that -> no extra check needed

Overall I think the only relevant capability is whether the user is allowed to upload media.
Typically the contributor role doesn't have that capability (and subscribers, but they can't even edit stories)

-----

NB: Happy to talk more about this and explain it a bit more in a call if needed.

Reply:
Much easier to digest when thinking about permissions rather than user roles! I'll update those 😄